services/admin/frontend/Modal/Dialog.svelte

```diff
@@ -0,0 +1,101 @@
+<script>
+    import {getContext, onMount} from 'svelte';
+    import {createProject, editProject, currentProject} from "../store";
+
+    export let onCancel = () => {};
+    export let onOkay = () => {};
+
+    export let editMode;
+
+    const { close } = getContext('simple-modal');
+
+    const projectID = window.location.pathname.split("/")[2];
+    let shortCode;
+    let shortName;
+    let longName;
+    let description;
+
+    let onChange = () => {};
+
+    function _onCancel() {
+        onCancel();
+        close();
+    }
+
+    async function _onOkay(){
+        if (editMode) {
+            await editProject(projectID, shortCode, shortName, longName, description);
+        } else {
+            await createProject(shortCode, shortName, longName, description);
+        }
+        close();
+    }
+
+    $: onChange(shortCode);
+    $: onChange(shortName);
+    $: onChange(longName);
+    $: onChange(description);
+
+    onMount(() => {
+        if (editMode) {
+            shortCode = $currentProject.shortCode;
+            shortName = $currentProject.shortName;
+            longName = $currentProject.longName;
+            description = $currentProject.description;
+        }
+    });
+</script>
+
+<style>
+    h2 {
+        font-size: 2rem;
+        text-align: center;
+    }
+
+    input {
+        width: 100%;
+    }
+
+    .buttons {
+        display: flex;
+        justify-content: space-between;
+    }
+</style>
```

Comment:
in Svelte it seems to be more common to have `<style>` at the end of the component

Reply:
done in b681dfc